CRM/Contact/BAO/GroupContact.php

```diff
@@ -239,6 +239,10 @@ public static function removeContactsFromGroup(
         CRM_Contact_BAO_SubscriptionHistory::create($historyParams);
         $groupContact->status = $status;
         $groupContact->save();
+        // Remove any rows from the group contact cache so it disappears straight away from smart groups.
+        CRM_Core_DAO::executeQuery('DELETE FROM civicrm_group_contact_cache WHERE group_id = %1 AND contact_id = %2', [
+          1 => [$groupId, 'Integer'], 2 => [$contactId, 'Integer']
+        ]);
```

Comment:
We already have a function to do this in `GroupContactCache.php` - `CRM_Contact_BAO_GroupContactCache::removeContact()` - can we use that please?

Reply:
thanks